### DISCUSSION TITLE
IBM網頁某些字型渲染沒有效果

### DISCUSSION THREAD

**Discussion by MrLuke1110** [F9y4ng/GreasyFork-Scripts#313] [问答专区(Question / Answer)] [answered]
再麻煩您確認，已試過重置參數，謝謝。

![image](https://github.com/F9y4ng/GreasyFork-Scripts/assets/66194793/4c90123a-3173-4196-886a-b792e68d0eb8)
![image](https://github.com/F9y4ng/GreasyFork-Scripts/assets/66194793/d0c3f04c-dd35-49b8-94f5-c445edbf8969)

**Comment by F9y4ng** [comment]
你觉得我能花多长时间猜出你访问的网页地址？

**Reply by MrLuke1110** [reply]
不好意思，忘了提供網址
https://www.ibm.com/support/pages/ibm-i-access-acs-updates

**Reply by F9y4ng** [reply]
该页面使用内联样式重新定义了部分字体，使用常规的字体设置会由于样式优先级的问题造成字体渲染无效。

但可以通过强制使用自定义等宽字体来提升优先级来实现，需要注意的是， 中文字体会继承使用常规设置，英文、数字使用等宽字体。

## 设置方法
1. 打开[自定义等宽字体工具](https://github.com/F9y4ng/GreasyFork-Scripts/wiki/%E5%AD%97%E4%BD%93%E6%B8%B2%E6%9F%93%EF%BC%88%E8%87%AA%E7%94%A8%E8%84%9A%E6%9C%AC%EF%BC%89#%E8%87%AA%E5%AE%9A%E4%B9%89%E7%AD%89%E5%AE%BD%E5%AD%97%E4%BD%93)；
2. 在文本域中添加规则`@www.ibm.com##.acs-content *`；
3. 可选填写自定义等宽字体及openType设置；
4. 保存数据即可。

**Comment by cxw620** [comment]
![image](https://github.com/F9y4ng/GreasyFork-Scripts/assets/70561268/d4f4846c-623c-4b2d-8f99-3ef59498324f)

把字体写死了

**Reply by F9y4ng** [reply, marked as answer]
遇到这种情况，可将脚本更新至v2024.03.02.1后，添加[字体重写规则](https://github.com/F9y4ng/GreasyFork-Scripts/wiki/%E5%AD%97%E4%BD%93%E6%B8%B2%E6%9F%93%EF%BC%88%E8%87%AA%E7%94%A8%E8%84%9A%E6%9C%AC%EF%BC%89#%E8%87%AA%E5%AE%9A%E4%B9%89%E5%AD%97%E4%BD%93%E9%87%8D%E5%86%99%E6%95%B0%E6%8D%AE):
```
"IBM Plex Mono",
"IBM Plex Sans",
```
保存后即可覆盖渲染字体。

**Reply by MrLuke1110** [reply]
成功運行無誤，感謝配合修正

**Reply by MrLuke1110** [reply]
不好意思，突然發現部份字體的效果，沒有被正確渲染，已有添加 IBM Plex Mono 至字體重寫規則，可以的話再麻煩確認，謝謝。
[https://www.ibm.com/support/pages/ibm-i-access-acs-updates](https://www.ibm.com/support/pages/ibm-i-access-acs-updates)
![image](https://github.com/user-attachments/assets/57236f6a-0dc1-4f3d-a38d-6fd0655d5205)
![image](https://github.com/user-attachments/assets/19eb492e-58e3-4f07-8ee8-b2c36c4055ac)

**Reply by F9y4ng** [reply]
> 不好意思，突然發現部份字體的效果，沒有被正確渲染，已有添加 IBM Plex Mono 至字體重寫規則，可以的話再麻煩確認，謝謝。
> [https://www.ibm.com/support/pages/ibm-i-access-acs-updates](https://www.ibm.com/support/pages/ibm-i-access-acs-updates)

你添加的自定义字体有没有正确填写PostScript名称，如果没有，一部分字体热替换会因此而失效。

可参考 #261 的方法查询字体属性参数，重新填写自定义字体数据。
